Moq.AutoMock/MockExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Moq.Language.Flow;
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Moq.AutoMock
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public static class MockExtensions
+    {
+        /// <summary>
+        /// Specifies a setup on the mocked type for a call to a void method. 
+        /// All parameters are filled with <see cref ="It.IsAny" /> according to the parameter's type.
+        /// </summary>
+        /// <remarks>
+        /// This may only be used on methods that are not overloaded.
+        /// </remarks>
+        /// <typeparam name="T">Type of the mock</typeparam>
+        /// <param name="mock">The mock</param>
+        /// <param name="methodName">The name of the expected method invocation.</param>
+        /// <exception cref="ArgumentNullException">When mock or methodName is null.</exception>
+        /// <exception cref="MissingMethodException">Thrown when no method with methodName is found.</exception>
+        /// <exception cref="AmbiguousMatchException">Thrown when more that one method matches the passed method name.</exception>
+        /// <returns></returns>
+        public static ISetup<T> SetupWithAny<T>(this Mock<T> mock, string methodName)
+            where T : class
+        {
+            if (mock is null)
+            {
+                throw new ArgumentNullException(nameof(mock));
+            }
+
+            if (methodName is null)
+            {
+                throw new ArgumentNullException(nameof(methodName));
```

Comment:
nitpick: I see stuff like this and think, "you didn't add any additional info or context, so you should have let the NullPtrException happen." Or add info or context...

Reply:
I would argue that this is better than a NullReferenceException. In this case it tells the caller that they made a mistake, while a null reference exception would indicate that there was an error within this library.
In this particular case I would suggest that it is better to explicitly not provide a custom error message, since doing so would replace the localized messages provided by the BCL.